[CI] EmployeeFlightServerTests testGetFlightInfoReturnsSingleEndpoint failing

**Build Scans:**
- [elasticsearch-intake #35978 / part3](https://gradle-enterprise.elastic.co/s/yqzbcxabflsqy)
- [elasticsearch-pull-request #127941 / part-3](https://gradle-enterprise.elastic.co/s/duwg4mmdyrc7y)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql-datasource-grpc:test" --tests "org.elasticsearch.xpack.esql.datasource.grpc.EmployeeFlightServerTests.testGetFlightInfoReturnsSingleEndpoint" -Dtests.seed=D4C0C687206A3B9A -Dtests.locale=ca-IT -Dtests.timezone=America/Blanc-Sablon -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.datasource.grpc.EmployeeFlightServerTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testGetFlightInfoReturnsSingleEndpoint'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: an empty collection
     but: <[An exception was thrown by io.grpc.netty.NettyServerHandler$5.operationComplete()]>
```

**Issue Reasons:**
- [main] 2 failures in test testGetFlightInfoReturnsSingleEndpoint (0.4% fail rate in 513 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -294,6 +294,9 @@ tests:
 - class: org.elasticsearch.repositories.azure.AzureBlobContainerRetriesTests
   method: testWriteBlobWithRetries
   issue: https://github.com/elastic/elasticsearch/issues/144025
+- class: org.elasticsearch.xpack.inference.integration.AuthorizationTaskExecutorIT
+  method: testCreatesChatCompletion_AndThenCreatesTextEmbedding
+  issue: https://github.com/elastic/elasticsearch/issues/138012
 - class: org.elasticsearch.upgrades.MappingSupplementaryCharacterRollingUpgradeIT
   method: testMappingWithSupplementaryCharacterFieldName {upgradedNodes=3}
   issue: https://github.com/elastic/elasticsearch/issues/144163
@@ -309,15 +312,27 @@ tests:
 - class: org.elasticsearch.test.apmintegration.MetricsApmIT
   method: testJvmMetrics {withOTel=false}
   issue: https://github.com/elastic/elasticsearch/issues/144167
-- class: org.elasticsearch.xpack.esql.datasource.grpc.EmployeeFlightServerTests
-  method: testMultiEndpointEachPartitionHasRows
-  issue: https://github.com/elastic/elasticsearch/issues/144244
+- class: org.elasticsearch.xpack.esql.CsvTests
+  method: test {csv-spec:approximation.Approximate stats by on large multi-valued data}
+  issue: https://github.com/elastic/elasticsearch/issues/144169
+- class: org.elasticsearch.xpack.esql.CsvIT
+  method: test {csv-spec:approximation.Approximate stats by with zero variance}
+  issue: https://github.com/elastic/elasticsearch/issues/144240
+- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
+  method: test {csv-spec:approximation.Approximate stats by with zero variance}
+  issue: https://github.com/elastic/elasticsearch/issues/144246
+- class: org.elasticsearch.xpack.inference.integration.AuthorizationTaskExecutorIT
+  method: testEndpointGetsUpdated_GivenFingerprintChanges_FromNull
+  issue: https://github.com/elastic/elasticsearch/issues/144249
 - class: org.elasticsearch.xpack.security.authc.service.ServiceAccountIT
   method: testAuthenticateShouldNotFallThroughInCaseOfFailure
   issue: https://github.com/elastic/elasticsearch/issues/144262
-- class: org.elasticsearch.xpack.esql.datasource.grpc.EmployeeFlightServerTests
-  method: testGetFlightInfoReturnsSingleEndpoint
-  issue: https://github.com/elastic/elasticsearch/issues/144263
+- class: org.elasticsearch.xpack.esql.parser.StatementParserTests
+  method: testLimitBy
+  issue: https://github.com/elastic/elasticsearch/issues/144267
+- class: org.elasticsearch.xpack.esql.parser.StatementParserTests
+  method: testLimitByQualifiedName
+  issue: https://github.com/elastic/elasticsearch/issues/144268
 - class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
   method: test
   issue: https://github.com/elastic/elasticsearch/issues/143023

--- a/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java
@@ -336,8 +336,16 @@ public abstract class ESTestCase extends LuceneTestCase {
             @Override
             public void append(LogEvent event) {
                 if (Level.WARN.equals(event.getLevel())) {
+                    final String message = event.getMessage().getFormattedMessage();
+                    // gRPC's Netty transport can sometimes throw from an internal ChannelFutureListener during stream shutdown,
+                    // which Netty logs as a WARN on DefaultPromise:
+                    // "An exception was thrown by io.grpc.netty.NettyServerHandler$X.operationComplete()"
+                    // This is a known source of test flakiness for Arrow Flight based tests and is not actionable here.
+                    if (message.contains("An exception was thrown by io.grpc.netty.NettyServerHandler")) {
+                        return;
+                    }
                     synchronized (loggedLeaks) {
-                        loggedLeaks.add(event.getMessage().getFormattedMessage());
+                        loggedLeaks.add(message);
                     }
                 }
             }

--- a/x-pack/plugin/esql-datasource-grpc/src/test/java/org/elasticsearch/xpack/esql/datasource/grpc/EmployeeFlightServer.java
+++ b/x-pack/plugin/esql-datasource-grpc/src/test/java/org/elasticsearch/xpack/esql/datasource/grpc/EmployeeFlightServer.java
@@ -254,7 +254,8 @@ public class EmployeeFlightServer implements Closeable {
             // have pending ChannelFutureListener callbacks (e.g. NettyServerHandler.closeStreamWhenDone)
             // that can throw if the HTTP/2 stream was already closed. Netty's DefaultPromise logs
             // these as WARN which ESTestCase.checkStaticState() treats as a test failure.
-            // A brief sleep lets those callbacks drain on the event loop thread before the test ends.
+            // ESTestCase now filters this specific gRPC NettyServerHandler warning (see #144244, #144263).
+            // A brief sleep still helps reduce log noise.
             server.close();
             Thread.sleep(50);
         } catch (InterruptedException e) {

